chain/network/src/routing.rs

```diff
@@ -224,21 +262,61 @@ impl RoutingTable {
             edges_info: HashMap::new(),
             route_back: CloneNone::new(SizedCache::with_size(ROUTE_BACK_CACHE_SIZE)),
             raw_graph: Graph::new(peer_id),
+            route_nonce: HashMap::new(),
+            recalculation_scheduled: None,
             ping_info: None,
             pong_info: None,
         }
     }
 
     /// Find peer that is connected to `source` and belong to the shortest path
     /// from `source` to `peer_id`.
-    pub fn find_route_from_peer_id(&self, peer_id: &PeerId) -> Result<PeerId, FindRouteError> {
+    pub fn find_route_from_peer_id(&mut self, peer_id: &PeerId) -> Result<PeerId, FindRouteError> {
         if let Some(routes) = self.peer_forwarding.get(&peer_id) {
-            if routes.is_empty() {
-                Err(FindRouteError::Disconnected)
-            } else {
-                // TODO(MarX, #1363): Do Round Robin
-                Ok(routes.iter().next().unwrap().clone())
-            }
+            // Strategy similar to Round Robin. Select node with least nonce and send it. Increase its
+            // nonce by one. Additionally if the difference between the highest and nonce and the lowest
+            // nonce is greater than some threshold increase the lowest nonce to be at least
+            // max nonce - threshold.
+
+            let (min_v, max_v) = routes
+                .iter()
+                .map(|peer_id| {
+                    let nonce = self.route_nonce.get(&peer_id).cloned().unwrap_or(0usize);
+                    (nonce, peer_id.clone())
+                })
+                .fold((None, None), |(min_v, max_v), current| {
```

Comment:
Why not fold directly?

Reply:
No need to, that was an old piece of code refactorized. I'll remove `map`.